### PR TITLE
Use eglGetPlatformDisplayEXT for ARM Mali Midgard DRM drivers.

### PR DESCRIPTION
**Pull request by Miouyouyou**
When using ARM Mali Midgard Wayland/DRM drivers r12p0, eglGetDisplay
does not work in DRM mode. Only eglGetPlatformDisplayEXT works.

There's still some issues unresolved, though, with these drivers. 
Currently, the benchmarks only work when run like this:

    glmark2-es-drm --visual-config alpha=0:stencil=1

Also removed the code testing each possible DRM driver. Manual copy
of @elmarco code fix.

Signed-off-by: Myy <myy@miouyouyou.fr>

### CODE REVIEW

File: src/native-state-drm.cpp [hunk 1]
```diff
@@ -197,27 +199,7 @@ NativeStateDRM::init_gbm()
...
-    for (unsigned int m = 0; m < num_modules; m++) {
-        fd_ = drmOpen(drm_modules[m], 0);
-        if (fd_ < 0) {
-            Log::debug("Failed to open DRM module '%s'\n", drm_modules[m]);
-            continue;
-        }
-        Log::debug("Opened DRM module '%s'\n", drm_modules[m]);
-        break;
-    }
+    fd_ = open("/dev/dri/card0", O_RDWR);
```
Comment (fooishbar) [inline src/native-state-drm.cpp]: This is unrelated, and will break when running with, e.g., vgem. I guess you just need to add Mali to the whitelist.

Comment (Miouyouyou) [inline src/native-state-drm.cpp]: If the `vgem` driver does not expose a `/dev/dri/card0` device node, then either the node is already defined by another driver, in which the `vgem` node will be `/dev/dri/card1`, or the driver should be fixed, IMHO.

The whitelist system should be avoided at all costs, as you can't maintain a hardcoded list of past, present, future and exotic DRM modules.

If the DRM driver does not expose `/dev/dri/card0` anymore, then either :
* the sysadmin used udev to rename the dev entry;
* another DRM driver has already created this device node, in which case `/dev/dri/card1` should be available;
* the device driver has serious issues;
* the standard completely changed.

I don't know where this whitelist system comes from. KMS/DRM tutorials clearly define that you should just open `/dev/dri/card0` : [Writing standalone programs with EGL and KMS (updated)](http://virtuousgeek.org/blog/index.php/jbarnes/2011/10/31/writing_stanalone_programs_with_egl_and_)

This is also how the MESA guys do it now : https://github.com/robclark/kmscube/blob/master/kmscube.c#L76

That said, I can :
* split the patch in two,
* use this whitelist system as a fallback mechanism, for old device drivers that do not expose /dev/dri/card0,
* provide a way to define the device node to use for DRM drivers, for multiple graphic cards systems, like this here https://github.com/dvdhrm/docs/blob/master/drm-howto/modeset.c

Comment (fooishbar) [inline src/native-state-drm.cpp]: vgem does expose a card0 node, but as the name implies, it's a purely virtual driver. So it's entirely possible to bind vgem on card0 (useful for memory allocation/sharing and nothing else), and have your real hardware driver on card1. The tutorial on virtuousgeek.org is not correct in that sense.

Comment (Miouyouyou) [inline src/native-state-drm.cpp]: In that case, can the program interact the with this DRM driver by opening `/dev/dri/card1` ?

Comment (fooishbar) [inline src/native-state-drm.cpp]: @Miouyouyou Right, it can, but it would be even better if the user didn't have to pass paths which might change (e.g. if module load / device probe order changes)

Comment (Miouyouyou) [inline src/native-state-drm.cpp]: That I agree, detection should be done automatically.

I'm looking at [KWin DRM code](https://cgit.kde.org/kwin.git/tree/udev.cpp#n128). It seems they just use UDev to scan for render nodes, and [fall back on virtual rendering nodes if nothing can be done](https://cgit.kde.org/kwin.git/tree/plugins/platforms/virtual/egl_gbm_backend.cpp#n65).

Meanwhile, I'll split this patch in two.

**Comment by Miouyouyou** [comment]
Pull request re-initiated here https://github.com/glmark2/glmark2/pull/33
